thirdeye/thirdeye-frontend/app/pods/manage/alerts/index/controller.js

```diff
@@ -281,146 +297,50 @@ export default Controller.extend({
-    if (filters.primary) {
-      this._processPrimaryFilters(initialAlerts, filters.primary);
-    }
-    // Pick up cached alert array for the secondary filters
-    let filteredAlerts = get(this, 'filteredAlerts');
-    // If there is a secondary filter present, filter by it, using the keys we've set up in our filter map
-    Object.keys(filterToPropertyMap).forEach((filterKey) => {
-      let filterValueArray = filters[filterKey];
-      if (filterValueArray && filterValueArray.length) {
-        let newAlerts = filteredAlerts.filter(alert => {
-          // See 'filterToPropertyMap' in route. For filterKey = 'owner' this would map alerts by alert['createdBy'] = x
-          const targetAlertPropertyValue = alert[filterToPropertyMap[filterKey]];
-          let alertMeetsCriteria = false;
-          // In the cases for subscription and application, there can be multiple values.  We just need to match on one
-          if (filterKey === "subscription") {
-            if (targetAlertPropertyValue) {
-              filterValueArray.forEach(val => {
-                if (targetAlertPropertyValue.includes(val)) {
-                  alertMeetsCriteria = true;
-                }
-              });
-            }
-          } else if (filterKey === "application") {
-            if (targetAlertPropertyValue) {
-              filterValueArray.forEach(val => {
-                if (targetAlertPropertyValue.includes(val)) {
-                  alertMeetsCriteria = true;
-                }
-              });
-            }
-          } else {
-            alertMeetsCriteria = targetAlertPropertyValue && filterValueArray.includes(targetAlertPropertyValue);
-          }
-          const isMatchForNone = !alert.hasOwnProperty(filterToPropertyMap[filterKey]) && filterValueArray.includes('none');
-          return alertMeetsCriteria || isMatchForNone;
-        });
-        filteredAlerts = newAlerts;
-      }
-    });
-
-    // If status filter is present, we re-build the results array to contain only active alerts, inactive alerts, or both.
-    if (filters.status) {
-      const concatStatus = filters.status.length ? filters.status.join().toLowerCase() : 'active';
-      const requireAll = filters.status.includes('Active') && filters.status.includes('Inactive');
-      const alertsByState = {
-        active: filteredAlerts.filter(alert => alert.active),
-        inactive: filteredAlerts.filter(alert => !alert.active)
-      };
-      filteredAlerts = requireAll ? [ ...alertsByState.active, ...alertsByState.inactive ] : alertsByState[concatStatus];
+  mySubscriptionGroups: computed(
+    'subscriptionGroupConfig',
+    'user',
+    function() {
+      const {
+        user,
+        subscriptionGroupConfig
+      } = this.getProperties('user', 'subscriptionGroupConfig');
+
+      // Find subscription groups current user is associated with
+      return subscriptionGroupConfig.filter((group) => {
```

Comment:
For searching the alerts the user is subscribed to, there is a param in the `alerts` endpoint called `subscribedBy`. The UI can get the related alerts directly by passing the user name into endpoint.